Generate.py

```diff
@@ -256,18 +333,225 @@ def get_choice_legacy(option, root, value=None) -> Any:
+                if paren_count <= 0:
+                    raise ValueError(f"unmatched close parentheses at {i} in: {s}")
+                paren_count -= 1
+                tokens.append(CloseParenToken(")"))
+                i += 1
+            elif _is_bin_op(val_str := s[i]):
+                tokens.append(BinOpToken(val_str))
+                state = "start"
+                i += 1
+            else:
+                raise ValueError(f"unexpected symbol {s[i]} at {i} in: {s}")
+    if paren_count != 0:
+        raise ValueError(f"unclosed parentheses in: {s}")
+    if state != "end":
+        raise ValueError(f"invalid expression: {s}")
+    return tokens
+
+
+def infix_to_postfix(infix: Sequence[AllTokens]) -> Sequence[PostfixTokens]:
+    infix = [OpenParenToken("(")] + list(infix) + [CloseParenToken(")")]
+    op_stack: List[Union[OpenParenToken, BinOpToken]] = []
+    postfix: List[PostfixTokens] = []
+    for token in infix:
+        if isinstance(token, OpenParenToken):
+            op_stack.append(token)
+        elif isinstance(token, CloseParenToken):
+            while len(op_stack):
+                op = op_stack.pop()
+                if isinstance(op, OpenParenToken):
+                    break
+                else:
+                    assert isinstance(assert_type(op, BinOpToken), BinOpToken), f"{op=}"
+                    postfix.append(op)
+        elif isinstance(token, BinOpToken):
+            while True:
+                assert len(op_stack), f"{infix=}"
+                top = op_stack[-1]
+                if _precedence[top.val] >= _precedence[token.val]:
+                    top = op_stack.pop()
+                    assert isinstance(top, BinOpToken), f"{top=} {infix=}"
+                    postfix.append(top)
+                else:
+                    break
+            op_stack.append(token)
+        else:
+            assert isinstance(token, NumberToken) or isinstance(assert_type(token, NameToken), NameToken), f"{token=}"
+            postfix.append(token)
+    return postfix
+
+
+def eval_postfix(postfix: Sequence[PostfixTokens], name_resolver: Callable[[str], float]) -> float:
+    operand_stack: List[float] = []
+    for token in postfix:
+        if isinstance(token, NumberToken):
+            operand_stack.append(token.val)
+        elif isinstance(token, NameToken):
+            operand_stack.append(name_resolver(token.val))
+        else:
+            assert isinstance(assert_type(token, BinOpToken), BinOpToken), f"{token=}"
+            if len(operand_stack) < 2:
+                raise ValueError(f"invalid {postfix=}")
+            b = operand_stack.pop()
+            a = operand_stack.pop()
+            operand_stack.append(_bops[token.val](a, b))
+    if len(operand_stack) != 1:
+        raise ValueError(f"invalid {postfix=}")
+    return operand_stack[0]
+
+
+def get_choice(option, root, value=None, sub_group=None, record: ChoiceRecord = ChoiceRecord(None, None, None)) -> Any:
```

Comment:
This is a mutable default argument for `record`
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments